### ISSUE TITLE
Bar chart x-axis factor order wrong

### ISSUE THREAD

**Issue by 13bzhang** [ropensci/plotly#242]
When I was writing tests for #241, I came across this problem where the bar chart x-axis factor level order is wrong. 

R code:
```r
set.seed(23049)
long_items <- data.frame(cat1 = sample(x = LETTERS[1:10], 
                                  size = 100, replace = TRUE),
                    cat2 = sample(x = LETTERS[1:10],
                                  size = 100, replace = TRUE))
p <- ggplot(long_items, aes(cat1, fill=cat2)) + geom_bar(position="dodge")
py$ggplotly(p)
```

ggplot2 figure:
![abc_ggplot2](https://cloud.githubusercontent.com/assets/6131842/9005253/9e083e1c-374a-11e5-92f6-9759f6697f31.png)

Note the figure above has the x-axis in alphabetical order.

Plotly figure (based on PR #241): https://plot.ly/~13bzhang/1365/count-vs-cat1/
![abc_plotly_master](https://cloud.githubusercontent.com/assets/6131842/9005279/e2ee95bc-374a-11e5-83b2-bc93b93248a4.png)

Now the x-axis is not in alphabetical order.

@cpsievert @chriddyp @mkcor

**Comment by cpsievert** [comment]
The x-axis ordering is controlled by the order in which values first appear in the trace data:

```r
l <- plotly_build(p)
# notice how this matches the x-axis ordering
unique(unlist(lapply(l$data, "[[", "x")))
#>  [1] A B C E F I G H J D
#> Levels: A B C D E F G H I J
```

It seems like I should be able to control this ordering by setting the range of the axis, but this doesn't do what I want, any ideas @chriddyp @cldougl?

```r
l$layout$xaxis$range <- LETTERS[1:10]
l
```

**Comment by cpsievert** [comment]
This should be fixed in f1900cc3cd464408a60ffadfe3e95460747430e0. Let us know if you still have problems.
